todolist added, folder structure changed

kishan9192 opened the pull request.

kishan9192 committed:
refactor: changing folder structure

code refactor for following the organizational way of folder structure

ypahalajani commented on src/components/todolist/styles.css:
```diff
@@ -0,0 +1,46 @@
+.toast {
+  position: relative;
+  display: inline;
+  left: 990px;
```
Instead of hard coding the width, we can use `left: 100%` as a starting point for the toast to be just out of the viewport and have display: 'none' CSS so that it is not visible by default

ypahalajani replied:
Update the values accordingly in animation.

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+/* eslint-disable react-hooks/exhaustive-deps */
```
I hope there are no errors when removing these

kishan9192 replied:
Yes, this just gives some warnings. Like for unused variables. Because of ES-Lint extension

kishan9192 replied:
Yes no errors, these are warnings like for unused variables, because of ESLint extension

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+import "@innovaccer/design-system/css";
+import { MessageAppearance } from "@innovaccer/design-system/dist/core/common.type";
+import { Dispatch } from "react";
+import React, { useEffect, useState } from "react";
```
Imports can be merged. - Our tsconfig would prevent this from being committed.

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+import { MessageAppearance } from "@innovaccer/design-system/dist/core/common.type";
+import { Dispatch } from "react";
+import React, { useEffect, useState } from "react";
+import { addItem, deleteItem, editItem } from "../../reducer/todolist/action";
```
Convention: We leave an empty line to separate library imports from internal file imports

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+import { Formik, Form, Field, ErrorMessage, useFormik } from "formik";
+import "./styles.css";
+
+const TodoList = (props: PropType) => {
```
```suggestion
const TodoList: React.FC<IProps> = props => {
```

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+import "./styles.css";
+
+const TodoList = (props: PropType) => {
+  const [todoItems, setTodoItems] = useState<Array<ItemType>>([]);
```
A relatively simpler form
```suggestion
  const [todoItems, setTodoItems] = useState<ItemType[]>([]);
```

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+
+const TodoList = (props: PropType) => {
+  const [todoItems, setTodoItems] = useState<Array<ItemType>>([]);
+  const [isFormOpen, setIsFormOpen] = useState(false);
```
There's still code that is not being used anymore - remove the tslint disable comment at the top of this file

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+    id: "",
+  });
+
+  useEffect(() => {
```
What is the purpose of this?

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+
+  const CaptureTodo = () => {
+    const [todoValue, setTodoValue] = useState("");
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      setTodoValue(e.target.value);
+    };
```
A relatively simpler usage and less bulky code
```suggestion
    const onTextChange = (newText: string) => setTodoValue(newText);
```

Also, update the calling place to this
```
onChange={event => onTextChange(event.target.value)}
```

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+      <Modal
+        open={isFormOpen}
+        dimension="small"
+        backdropClose={true}
```
You can avoid setting explicitly true. Alternative syntax 👇 
```suggestion
        backdropClose
```

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+    );
+  };
+
+  const editTodo = (todo: ItemType) => {
+    setSelectedItem({ ...selectedItem, oldValue: todo.item, id: todo.id });
+  };
```
```suggestion
  const editTodo = (todo: ItemType) => setSelectedItem({ ...selectedItem, oldValue: todo.item, id: todo.id })
```

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+  const [actionPerformed, setActionPerformed] = useState(false);
+  const [selectedItem, setSelectedItem] = useState({
+    updatedValue: "",
+    oldValue: "",
```
You might not need `oldValue` at all to be managed in a state.

ypahalajani replied:
Let's discuss

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+                }}
+              >
+                <Text
+                  // className="flex-wrap"
```
Convention: we do not keep commented code in the codebase

ypahalajani commented on src/reducer/todolist/action.ts:
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Remove this line and remove unused code.

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+// need to import Dispatch from react, then pass the action type in Dispatch as type parameter
+function mapDispatchToProps(
+  dispatch: Dispatch<IDispatch>
+): IMapDispatchToProps {
```
I believe that the TS should be smart enough to infer the return type
```suggestion
) {
```

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+        >
+          <DisplayTodos />
+        </div>
+        {actionPerformed && showToast()}
```
```suggestion
        {showToast && renderToast()}
```

OR 

Create a component that uses your implementation of Toast
```suggestion
        {showToast && <Toast />}
```

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -0,0 +1,292 @@
+  const [todoItems, setTodoItems] = useState<Array<ItemType>>([]);
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [toastType, setToastType] = useState<MessageAppearance>();
+  const [actionPerformed, setActionPerformed] = useState(false);
```
Whether to show a toast or not is mostly global in an application, where you expose the actions to dispatch which can show/hide the toast.
Move this to redux store as well.

kishan9192 committed:
refactor (Yup|Toast-Redux|ellipsis) ui-improvements

changes in form, added validations using Yup, moving toast states to redux store and other ui-enhancements

ypahalajani commented on package.json:
```diff
@@ -18,11 +18,14 @@
     "typescript": "^4.5.4",
     "typescript-fsa": "^3.0.0",
     "typescript-fsa-reducers": "^1.2.2",
+    "typescript-fsa-redux-thunk": "^2.10.2",
```
Remove if not being used

ypahalajani commented on src/components/todolist/styles.css:
```diff
@@ -1,41 +1,24 @@
   }
   100% {
-    left: 990px;
+    left: 100%;
```
from `left: 100%;` to `right: 0%;`

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -1,109 +1,87 @@
+const TodoList: React.FC<IProps> = (props) => {
+  const [selectedID, setSelectedID] = useState("");
+  const [modalOpen, setModalOpen] = useState(false);
+  const [modalContent, setModalContent] = useState("");
```
const [modalState, setModalState] = useState({ open, content })

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -1,109 +1,87 @@
+    const element = document.getElementById(htmlElementID);
+    if (element && element.offsetWidth < element.scrollWidth) {
+      setModalOpen(true);
+      setModalContent(element?.innerText);
```
```suggestion
      setModalContent(element?.innerText || "");
```

ypahalajani commented on src/interfaces/entities/todolist/index.ts:
```diff
@@ -13,18 +14,33 @@ export interface IMapStateToProps {
+    duration: number
+}
+export interface IState {
+    todos: ITodosState,
```
`ITodoItem[]`

ypahalajani commented on src/components/todolist/todolist.tsx:
```diff
@@ -245,47 +238,49 @@ const TodoList = (props: PropType) => {
-): IMapDispatchToProps {
+function mapDispatchToProps(dispatch: Dispatch<IDispatch>) {
   return {
     addTodoDispatch: (payload: string) => dispatch(addItem(payload)),
```
Try it
```suggestion
    addTodoDispatch: addItem,
```

kishan9192 replied:
Not giving any error, but also not dispatching the action

kishan9192 committed:
refactor (redux-thunk): ux-improvements

changes in rendering list item, added redux-thunk, changes in variables and function names

kishan9192 committed:
tests: adding tests

Adding new tests for todolist and listItem using jest, react-testing-library and enzyme
